added migration guide for Bolt@2.x (#442)
slackapi/bolt: docs/_advanced/error_handling.md

---

<div class="section-content">

> *Note: Since `Bolt@2.x`, error handling has improved! View the [migration guide for V2](https://slack.dev/bolt/tutorial/migration-v2) to learn more.*

Reviewer: i think the `Bolt@2.x` naming is new and possibly unfamiliar. can we just say v2? if we're going to be specific enough to put it in a code format, then it should probably be the package/version name: `@slack/bolt@v2.0.0`

Reviewer: ```suggestion
*Note: Since v2, error handling has improved! View the [migration guide for V2](https://slack.dev/bolt/tutorial/migration-v2) to learn about the changes.*
```

> If an error occurs in a listener, it’s recommended you handle it directly with a `try`/`catch`. However, there still may be cases where errors slip through the cracks. By default, these errors will be logged to the console. To handle them yourself, you can attach a global error handler to your app with the `error(fn)` method.

Reviewer: ```suggestion
If an error occurs in a listener, it’s recommended you handle it directly with a `try`/`catch`. However, there still may be cases where errors slip through the cracks. By default, these errors will be logged to the console. To handle them yourself, you can attach a global error handler to your app with the `app.error(fn)` method.
```
